src/Model/RegisteredMethod.php

```diff
@@ -29,24 +30,24 @@ class AuthenticationMethod extends DataObject
     ];
 
     /**
-     * @var AuthenticationMethodInterface
+     * @var MethodInterface
      */
     protected $method;
 
     /**
-     * @return AuthenticatorInterface
+     * @return LoginHandlerInterface
      */
-    public function getAuthenticator()
+    public function getLoginHandler()
     {
-        return $this->getMethod()->getAuthenticator();
+        return $this->getMethod()->getLoginHandler();
     }
 
     /**
-     * @return mixed
+     * @return RegisterHandlerInterface
      */
-    public function getRegistrar()
+    public function getRegisterHandler()
     {
-        return $this->getMethod()->getRegistrar();
+        return $this->getMethod()->getRegisterHandler();
```

Comment:
These two methods have null pointer exception possibilities but we can fix that later